shared/chat/conversation/list/index.desktop.js

```diff
@@ -269,27 +269,41 @@ class PopupEnabledList extends BaseList {
       return !!(m && m.type === 'Text' && m.author === this.props.you)
     })
 
-    if (entry) {
-      const idx: number = entry[0]
-      const messageKey: Constants.MessageKey = entry[1]
-      // $ForceType
-      const message: Constants.TextMessage = this.props.getMessageFromMessageKey(messageKey)
-
-      this._keepIdxVisible = idx
-      this.setState({listRerender: this.state.listRerender + 1})
-
-      const listNode = ReactDOM.findDOMNode(this._list)
-      if (listNode) {
-        // $FlowIssue
-        const messageNodes = listNode.querySelectorAll(`[data-message-key="${messageKey}"]`)
-        if (messageNodes) {
-          const messageNode = messageNodes[0]
-          if (messageNode) {
-            this._showEditor(message, this._domNodeToRect(messageNode))
-          }
-        }
-      }
+    if (!entry) {
+      return
+    }
+
+    const idx: number = entry[0]
+    const messageKey: Constants.MessageKey = entry[1]
+    // $ForceType
+    const message: Constants.TextMessage = this.props.getMessageFromMessageKey(messageKey)
+
+    if (message.messageState !== 'sent') {
+      // For now, disallow editing of non-sent messages. In the
+      // future, we may want to do something more intelligent.
```

Comment:
Currently, there are some race conditions if you hit up immediately after entering a message; either nothing happens at all, or you might edit the previous message. Not worth worrying about at this point, though.